NitroxClient/GameLogic/Spawning/Bases/BuildingPostSpawner.cs

```diff
@@ -28,66 +31,75 @@ public static IEnumerator ApplyPostSpawner(GameObject gameObject, NitroxId objec
-
-                GameObject benchTileCenter = new("BenchPlaceCenter") { layer = LAYER_USEABLE };
-                benchTileCenter.transform.SetParent(gameObject.transform, false);
-                BoxCollider benchTileCenterCollider = benchTileCenter.AddComponent<BoxCollider>();
-                benchTileCenterCollider.center = new Vector3(0, 0.25f, 0);
-                benchTileCenterCollider.size = new Vector3(0.7f, 0.5f, 0.65f);
-                benchTileCenterCollider.isTrigger = true;
-
-                GameObject benchTileRight = new("BenchPlaceRight") { layer = LAYER_USEABLE };
-                benchTileRight.transform.SetParent(gameObject.transform, false);
-                benchTileRight.transform.localPosition += new Vector3(0.75f, 0, 0);
-                BoxCollider benchTileRightCollider = benchTileRight.AddComponent<BoxCollider>();
-                benchTileRightCollider.center = new Vector3(0, 0.25f, 0);
-                benchTileRightCollider.size = new Vector3(0.85f, 0.5f, 0.65f);
-                benchTileRightCollider.isTrigger = true;
-
-                GameObject animationRoot = gameObject.FindChild("bench_animation");
-
-                MultiplayerBench.FromBench(bench, benchTileLeft, MultiplayerBench.Side.LEFT, animationRoot);
-                MultiplayerBench.FromBench(bench, benchTileCenter, MultiplayerBench.Side.CENTER, animationRoot);
-                MultiplayerBench.FromBench(bench, benchTileRight, MultiplayerBench.Side.RIGHT, animationRoot);
-
-                NitroxId benchLeftId = benchId.Increment();
-                NitroxId benchCenterId = benchLeftId.Increment();
-                NitroxId benchRightId = benchCenterId.Increment();
-                NitroxEntity.SetNewId(benchTileLeft, benchLeftId);
-                NitroxEntity.SetNewId(benchTileCenter, benchCenterId);
-                NitroxEntity.SetNewId(benchTileRight, benchRightId);
-
-                UnityEngine.Object.Destroy(bench);
-                UnityEngine.Object.Destroy(gameObject.FindChild("Builder Trigger"));
-            }
-            catch (Exception ex)
-            {
-                Log.Error(ex);
-            }
+            Log.Error(ex);
         }
-        yield break;
     }
 
-    public static IEnumerator SetupLargeRoomWaterPark(LargeRoomWaterPark baseModule, NitroxId waterParkId)
+    public static void SetupWaterPark(WaterPark waterPark, NitroxId waterParkId)
     {
-        NitroxId leftId = waterParkId.Increment();
-        NitroxId rightId = leftId.Increment();
+        if (waterPark is LargeRoomWaterPark largeRoomWaterPark)
+        {
+            NitroxId leftId = waterParkId.Increment();
+            NitroxId rightId = leftId.Increment();
+
+            NitroxEntity.SetNewId(largeRoomWaterPark.planters.leftPlanter.gameObject, leftId);
+            NitroxEntity.SetNewId(largeRoomWaterPark.planters.rightPlanter.gameObject, rightId);
+            return;
+        }
+
+        NitroxId planterId = waterParkId.Increment();
```

Comment:
I don't have the increment code in front of me but I wanted to check this gives not the same as the other waterpark increment call.

Reply:
This is a very precise calculation which always adds 1 to the hexadecimal in the NitroxId's GUID. It has to be this deterministic (and not less) so we can actually pretend to have sync here